Unknown issue in role reset

When submitting the modal, something breaks:
```

18:34:52.715 [DefaultDispatcher-worker-1] ERROR io.github.populus_omnibus.vikbot.VikBotHandler -- Error while handling ModalInteractionEvent(id=1185288884322566354): 50035: Invalid Form Body
components[0].components[3].custom_id
	- COMPONENT_CUSTOM_ID_DUPLICATED: Component custom id cannot be duplicated
net.dv8tion.jda.api.exceptions.ErrorResponseException: 50035: Invalid Form Body
components[0].components[3].custom_id
	- COMPONENT_CUSTOM_ID_DUPLICATED: Component custom id cannot be duplicated
	at net.dv8tion.jda.internal.requests.RestActionImpl.complete(RestActionImpl.java:227)
	at net.dv8tion.jda.api.requests.RestAction.complete(RestAction.java:632)
	at io.github.populus_omnibus.vikbot.bot.modules.roleselector.RoleReset$init$2$1.invoke(RoleReset.kt:75)
	at io.github.populus_omnibus.vikbot.bot.modules.roleselector.RoleReset$init$2$1.invoke(RoleReset.kt:45)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:275)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.access$inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:1)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$inTopLevelTransaction$1.invoke(ThreadLocalTransactionManager.kt:322)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:330)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:321)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$transaction$1.invoke(ThreadLocalTransactionManager.kt:230)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:330)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:200)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:178)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:177)
	at io.github.populus_omnibus.vikbot.bot.modules.roleselector.RoleReset$init$2.invokeSuspend(RoleReset.kt:45)
	at io.github.populus_omnibus.vikbot.bot.modules.roleselector.RoleReset$init$2.invoke(RoleReset.kt)
	at io.github.populus_omnibus.vikbot.bot.modules.roleselector.RoleReset$init$2.invoke(RoleReset.kt)
	at io.github.populus_omnibus.vikbot.api.interactions.IdentifiableInteractionHandler.invoke$suspendImpl(IdentifiableInteractionHandler.kt:12)
	at io.github.populus_omnibus.vikbot.api.interactions.IdentifiableInteractionHandler.invoke(IdentifiableInteractionHandler.kt)
	at io.github.populus_omnibus.vikbot.api.interactions.IdentifiableInteractionHandlerKt.invoke(IdentifiableInteractionHandler.kt:41)
	at io.github.populus_omnibus.vikbot.VikBotHandler$onEvent$2.invokeSuspend(VikBotHandler.kt:169)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: net.dv8tion.jda.api.exceptions.ContextException: null
	at net.dv8tion.jda.api.exceptions.ContextException.here(ContextException.java:54)
	at net.dv8tion.jda.api.requests.Request.<init>(Request.java:77)
	at net.dv8tion.jda.api.requests.RestFuture.<init>(RestFuture.java:40)
	at net.dv8tion.jda.internal.requests.RestActionImpl.submit(RestActionImpl.java:209)
	at net.dv8tion.jda.internal.requests.RestActionImpl.complete(RestActionImpl.java:219)
	... 28 common frames omitted
```


To avoid confusion, #102 does __not__ fix this, that is just an incorrect state recovery.

--- a/bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/roleselector/RoleReset.kt
+++ b/bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/roleselector/RoleReset.kt
@@ -53,10 +53,11 @@ object RoleReset :
                     }
                 }
 
+                var i = 0
                 val buttons = buttonData.map {
                     listOf(
                         Button.danger("rolereset:${it.first}", it.second),
-                        Button.secondary("separator", "|").asDisabled()
+                        Button.secondary("separator${i++}", "|").asDisabled()
                     )
                 }.flatten()
                 if (buttons.size < 2) return@transaction

--- a/bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/roleselector/RoleSelectorCommands.kt
+++ b/bot/src/main/kotlin/io/github/populus_omnibus/vikbot/bot/modules/roleselector/RoleSelectorCommands.kt
@@ -165,7 +165,12 @@ object RoleSelectorCommands :
                 }
 
                 val previous = transaction { group.lastPublished }?.let {
-                    event.guild!!.getTextChannelById(it.channelId)?.retrieveMessageById(it.messageId)?.complete()
+                    try {
+                        event.guild!!.getTextChannelById(it.channelId)?.retrieveMessageById(it.messageId)?.complete()
+                    } catch (t: Throwable) {
+                        logger.warn("Failed to delete previous role reset message", t)
+                        null
+                    }
                 }
 
 

